tests/syntax-tests/test_custom_assets.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+set -o errexit -o nounset -o pipefail
+
+test_file="tests/syntax-tests/source/BatTestCustomAssets/NoColorsUnlessCustomAssetsAreUsed.battestcustomassets"
+
+# First make sure our test syntax is not part of integrated syntaxes
+echo "Testing that BatTestCustomAssets is an unknown syntax:"
+bat -f --language BatTestCustomAssets "${test_file}" &&
+    echo "should have failed because of unknown syntax" &&
+    exit 1
+
+# Now build custom assets and include our test syntax
+# We do as instructed to regular users at
+# https://github.com/sharkdp/bat/blob/master/README.md#adding-new-syntaxes--language-definitions
+
+# Create dir to put our test syntax
+custom_syntaxes_dir="$(bat --config-dir)/syntaxes"
+mkdir -p "${custom_syntaxes_dir}"
+
+# Put the syntax in place
+cp tests/syntax-tests/BatTestCustomAssets.sublime-syntax "${custom_syntaxes_dir}/BatTestCustomAssets.sublime-syntax"
+
+# Build custom assets to include the above syntax
+echo "Building custom assets to include the BatTestCustomAssets syntax"
+bat cache --build
+
+# Now bat shall not fail when the syntax is used. If it does not fail, we can be reasonably sure
+# that the custom assets functionality is working
+echo "Testing that BatTestCustomAssets is a KNOWN syntax:"
+bat -f --language BatTestCustomAssets "${test_file}" ||
+    (echo "command shall not have failed this time, but did" &&
+        exit 1)
+
```

Comment:
I just realised it would also be a good idea to sanity check an existing syntax, to verify that the custom syntax is not the ONLY syntax that is now available (i.e. verify that `include_integrated_assets` work)

Reply:
@sharkdp Would you mind doing a high level code review of this PR before I do the above?

Reply:
This looks great - thank you! In principle, we would like to run this on every supported platform, right? (not sure if we can run bash scripts on Windows)

Reply:
Not sure about from CI, but "Git Bash for Windows" is a thing  :) (obviously the "git" part isn't so relevant here)

Reply:
Running bash on Windows is no issue, because we already do it a lot. The entire 'Create tarball' step in CI is written in bash for example, but runs just fine on Windows (`windows-2019` to be precise).

In principle, yes we probably should run this in the matrix. But I think the risk of windows-specific problems is low enough to postpone looking into that.